src/MetaMorpho.sol

```diff
@@ -376,6 +372,12 @@ contract MetaMorpho is ERC4626, ERC20Permit, Ownable2Step, Multicall, IMetaMorph
 
             if (supplyCap == 0) revert ErrorsLib.UnauthorizedMarket(id);
 
+            if (allocation.assets == type(uint256).max) {
+                allocation.assets = totalWithdrawn.zeroFloorSub(totalSupplied);
+
+                if (allocation.assets == 0) continue;
```

Comment:
```suggestion
                if (allocation.assets == 0) break;
```
If `totalSupplied` is greater than `totalWithdrawn` we should exit the supply loop

Reply:
I disagree: it can happen that `totalWithdrawn <= totalSupplied` due to the order of markets and how the state of liquidity changed in each market
`totalWithdrawn` & `totalSupplied` at this stage of the algorithm only include the partial sum of withdrawn assets (resp. supplied) up to the current market of the allocation, not the overall total!

Reply:
This reasoning will not hold anymore with #307 where markets can be withdrawn from at any time during the reallocation!

Reply:
Yes, let's not do the change